src/market.py

```diff
@@ -0,0 +1,209 @@
+"""Market module to interact with Serum DEX."""
+import base64
+from struct import Struct
+from typing import Any, NamedTuple
+
+from construct import Padding  # type: ignore
+from construct import Struct as cStruct  # type: ignore
+from solana.publickey import PublicKey
+from solana.rpc.api import Client
+
+from .slab import SLAB_LAYOUT
+
+DEFAULT_DEX_PROGRAM_ID = PublicKey(
+    "4ckmDgGdxQoPDLUkDT3vHgSAkzA3QRdNq5ywwY4sUSJn",
+)
+
+_MARKET_FORMAT = ""
+_MARKET_FORMAT += "<"  # little endian
+_MARKET_FORMAT += "5s"  # 5 bytes padding
+_MARKET_FORMAT += "8s"  # 8 bytes of accountFlag, treat it as string
+_MARKET_FORMAT += "32s"  # 32 bytes of ownAddress
+_MARKET_FORMAT += "Q"  # 8 bytes of vaultSignerNonce
+_MARKET_FORMAT += "32s"  # 32 bytes of baseMint
+_MARKET_FORMAT += "32s"  # 32 bytes of quoteMint
+_MARKET_FORMAT += "32s"  # 32 bytes of baseVault
+_MARKET_FORMAT += "Q"  # 8 bytes of baseDepositsTotal
+_MARKET_FORMAT += "Q"  # 8 bytes of baseFeesAccrued
+_MARKET_FORMAT += "32s"  # 32 bytes quoteVault
+_MARKET_FORMAT += "Q"  # 8 bytes of quoteDepositsTotal
+_MARKET_FORMAT += "Q"  # 8 bytes of quoteFeesAccrued
+_MARKET_FORMAT += "Q"  # 8 bytes of quoteDustThreshold
+_MARKET_FORMAT += "32s"  # 32 bytes requestQueue
+_MARKET_FORMAT += "32s"  # 32 bytes eventQueue
+_MARKET_FORMAT += "32s"  # 32 bytes bids
+_MARKET_FORMAT += "32s"  # 32 bytes asks
+_MARKET_FORMAT += "Q"  # 8 bytes of baseLotSize
+_MARKET_FORMAT += "Q"  # 8 bytes of quoteLotSize
+_MARKET_FORMAT += "Q"  # 8 bytes of feeRateBps
+_MARKET_FORMAT += "7s"  # 7 bytes padding
+
+_MINT_LAYOUT = ""
+_MINT_LAYOUT += "36s"
+_MINT_LAYOUT += "b"
+_MINT_LAYOUT += "3s"
+
+
+ORDER_BOOK_LAYOUT = cStruct(Padding(5), "account_flag" / Padding(4), "slab_layout" / SLAB_LAYOUT)
+
+
+class MarketState(NamedTuple):
+    """Market State to stored the parsed market data."""
```

Comment:
Let's make this the entirely deserialized `MarketState` so instead of `str` let's use `PublicKey`. And ` account_flags` should be an object (dictionary for now).